src/App.tsx

```diff
@@ -1,14 +1,47 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [isFocused, setIsFocused] = useState(false);
+  const [query, setQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const filteredPeople = useMemo(() => {
+    if (!appliedQuery.startsWith(' ')) {
+      return peopleFromServer
+        .filter(person => person.name.toLowerCase()
+          .includes(appliedQuery.toLowerCase()));
+    }
+
+    return peopleFromServer;
+  }, [appliedQuery]);
+
+  const onSelected = (person: Person) => {
+    setSelectedPerson(person);
+    setQuery(person.name);
+    setIsFocused(false);
+  };
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, 1000),
+    [],
+  );
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
Would be nice to hide the list of suggestions on text change according to the requirements